app/src/main/java/org/dieschnittstelle/mobile/android/skeleton/MainActivity.java

```diff
@@ -1,34 +1,93 @@
+import com.google.android.material.snackbar.Snackbar;
+
+import org.dieschnittstelle.mobile.android.skeleton.databinding.ActivityMainBinding;
+import org.dieschnittstelle.mobile.android.skeleton.databinding.ActivityTaskDetailViewBinding;
+import org.dieschnittstelle.mobile.android.skeleton.model.IUserDatabaseOperation;
+import org.dieschnittstelle.mobile.android.skeleton.model.User;
+import org.dieschnittstelle.mobile.android.skeleton.viewmodel.MainViewModel;
+import org.dieschnittstelle.mobile.android.skeleton.viewmodel.TaskListViewModel;
 
 public class MainActivity extends AppCompatActivity {
 
     private TextView welcomeText;
-    private FloatingActionButton showTaskListAction;
+    private EditText emailAdress;
+    private EditText password;
+    private Button showTaskListAction;
+    private MainViewModel viewModel;
+    private IUserDatabaseOperation userDBOperation;
+    private User user = new User("", "");
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        viewModel = new ViewModelProvider(this).get(MainViewModel.class);
+
+        ActivityMainBinding MainViewBinding = DataBindingUtil.setContentView(
+                this,
+                R.layout.activity_main
+        );
+        MainViewBinding.setMainViewModel(this.viewModel);
+        MainViewBinding.setLifecycleOwner(this);
+
+        userDBOperation = ((TaskApplication) getApplication()).getUserDatabaseOperation();
+        viewModel.setUserDBOperation(userDBOperation);
+
+        viewModel.setUser(user);
+
+        userDbInitialise(); // insert testdata
 
         welcomeText = findViewById(R.id.welcomeText);
         welcomeText.setText(R.string.welcome_message);
-        welcomeText.setOnClickListener(view -> this.showTaskListView());
 
-        showTaskListAction = findViewById(R.id.showTaskListAction);
-        showTaskListAction.setOnClickListener(view -> this.showTaskListView());
+        emailAdress = findViewById(R.id.editTextTextEmailAddress);
+        password = findViewById(R.id.editTextTextPassword);
+
+        // Login
+        this.viewModel.getLoginLiveData().observe(this, onLogin -> {
+            if (onLogin == "usernameIsNotAnEMail") {
+                showPersistentMessage(getString(R.string.login_username_error));
+            } else if (onLogin == "userAuthenticationIsFailed"){
+                showPersistentMessage(getString(R.string.login_authentication_failed));
+            } else if (onLogin == "passwordIsNotAllowed"){
+                showPersistentMessage(getString(R.string.login_password_not_allowed));
+            } else if (onLogin == "userIsAuthenticated") {
+                Intent callTaskOverviewIntent = new Intent(this, TaskListViewActivity.class);
+                startActivity(callTaskOverviewIntent);
+                this.finish();
+            }
+        });
```

Comment:
siehe `TaskListViewActivity` Klasse für `ProcessingState` als Referenz

```suggestion
        viewModel.getLoginState().observe(this, this::handleUserLoginState);
```